va/va.go

```diff
@@ -375,55 +376,36 @@ func (va *ValidationAuthorityImpl) validateChallenge(ctx context.Context, identi
 // channel as-is.
 func (va *ValidationAuthorityImpl) performRemoteValidation(
 	ctx context.Context,
-	domain string,
-	challenge core.Challenge,
-	authz core.Authorization,
+	req *vapb.PerformValidationRequest,
 	results chan *remoteValidationResult) {
 	for _, i := range rand.Perm(len(va.remoteVAs)) {
 		remoteVA := va.remoteVAs[i]
 		go func(rva RemoteVA, index int) {
-			_, err := rva.PerformValidation(ctx, domain, challenge, authz)
-			if err != nil {
-				// returned error can be a nil *probs.ProblemDetails which breaks the
-				// err != nil check so do a slightly more complicated unwrap check to
-				// make sure we don't choke on that.
-				// TODO(@cpu): Clean this up once boulder issue 2254[0] is resolved
-				// [0] https://github.com/letsencrypt/boulder/issues/2254
-				if p, ok := err.(*probs.ProblemDetails); ok && p != (*probs.ProblemDetails)(nil) {
-					// If the non-nil err was a non-nil *probs.ProblemDetails then we can
-					// log it at an info level. It's a normal non-success validation
-					// result and the remote VA will have logged more detail.
-					va.log.Infof("Remote VA %q.PerformValidation returned problem: %s", rva.Address, err)
-				} else if ok && p == (*probs.ProblemDetails)(nil) {
-					// If the non-nil err was a nil *probs.ProblemDetails then we don't need to do
-					// anything. There isn't really an error here.
-					err = nil
-				} else if canceled.Is(err) {
-					// If the non-nil err was a canceled error, ignore it. That's fine it
-					// just means we cancelled the remote VA request before it was
-					// finished because we didn't care about its result.
-					err = nil
-				} else if !ok {
-					// Otherwise, the non-nil err was *not* a *probs.ProblemDetails and
-					// was *not* a context cancelleded error and represents something that
-					// will later be returned as a server internal error
-					// without detail if the number of errors is >= va.maxRemoteFailures.
-					// Log it at the error level so we can debug from logs.
-					va.log.Errf("Remote VA %q.PerformValidation failed: %s", rva.Address, err)
-				}
-			}
 			result := &remoteValidationResult{
 				VAHostname: rva.Address,
 			}
-			if err == nil {
-				results <- result
-			} else if prob, ok := err.(*probs.ProblemDetails); ok {
-				result.Problem = prob
-				results <- result
-			} else {
+			res, err := rva.PerformValidation(ctx, req)
+			if err != nil && canceled.Is(err) {
+				// If the non-nil err was a canceled error, ignore it. That's fine: it
+				// just means we cancelled the remote VA request before it was
+				// finished because we didn't care about its result.
+				va.log.Infof("Remote VA %q.PerformValidation canceled: %s", rva.Address, err)
```

Comment:
Roger. I added this log line because a) it seemed like useful symmetry, and b) I needed to do _something_ with `err` inside the block to make the compiler happy :D 

I'll make this into a Problem instead.

Reply:
I've done this in a separate change instead: https://github.com/letsencrypt/boulder/pull/5011